src/tools/wrapComponent.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+const wrapComponent = ({ name, className, type }) => {
+  const Component = props => {
+    const componentClass = cx(className, props.className);
+    return React.createElement(type, {
```

Comment:
Curious why `React.createElement()` is used instead of `<type {…props} className={!!componentClass ? componentClass : undefined} />` - Is it for the comment of `className` or something else? Also good to have more explanation of `wrapComponent()`. Thanks!

Reply:
`<type>` is invalid in React as it's lowercase 😞

Reply:
Yeah given `type` is a React component/node wondered if it should be uppercased, though I may be missing something.

Reply:
Can definitely rebind it to an uppercase variable in the scope, might help with the class name check too.